packages/firestore/src/core/sync_engine.ts

```diff
@@ -154,6 +155,9 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
   private limboResolutionsByTarget: {
     [targetId: number]: LimboResolution;
   } = {};
+  private readonly maxConcurrentLimboResolutions: number = DEFAULT_MAX_CONCURRENT_LIMBO_RESOLUTIONS;
+  /** The keys of documents whose limbo resolutions are enqueued. */
+  private limboListenQueue: DocumentKey[] = [];
   private limboDocumentRefs = new ReferenceSet();
```

Comment:
> variable names will be confusing because they have no counterpart for the opposite state 

That's an interesting perspective but I think the reasoning is kind of subtle: that because only active limbo resolutions can have a target the active-ness is self-explanatory. Would this have been self-evident to you a week ago?

My reasoning was that the prefix helps identify which variables are used in which state, and suggests which elements have to be modified as you transition between states. That is, this would have the effect of grouping these things together more than necessarily differentiating similar structures of the same type that might be used in the different states.

Reply:
Makes sense.  I've added "active" and "enqueued" to all relevant variable names throughout this PR.